### PR TITLE
Added token validation to batch requests.

### PR DESCRIPTION
**Pull request by xbits**
Tokens in batch requests where not being validated

### CODE REVIEW

File: lib/devise_token_auth/version.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module DeviseTokenAuth
-  VERSION = '1.1.3'.freeze
+  VERSION = '1.1.4'.freeze
```
Comment (MaicolBen) [inline lib/devise_token_auth/version.rb]: why this change?

File: app/models/devise_token_auth/concerns/user.rb [hunk 2]
```diff
@@ -147,7 +147,7 @@ def token_can_be_reused?(token, client)
       updated_at.to_time > Time.zone.now - DeviseTokenAuth.batch_request_buffer_throttle &&
 
       # ensure that the token is valid
-      DeviseTokenAuth::TokenFactory.valid_token_hash?(last_token)
+      DeviseTokenAuth::Concerns::User.tokens_match?(last_token, token)
```
Comment (MaicolBen) [inline app/models/devise_token_auth/concerns/user.rb]: This should be in the factory, it doesn't make any sense to call a user's method for this

File: lib/devise_token_auth/version.rb (hunk 1 above)
Comment (xbits) [reply lib/devise_token_auth/version.rb:4]: I guess it doesn't make sense. Not used to making contributions to public repos.

File: app/models/devise_token_auth/concerns/user.rb (hunk 2 above)
Comment (xbits) [reply app/models/devise_token_auth/concerns/user.rb:150]: Same method call is  used right in the previous method, and tokens_match? is defined only on the user.

File: lib/devise_token_auth/version.rb (hunk 1 above)
Comment (MaicolBen) [reply lib/devise_token_auth/version.rb:4]:
```suggestion
  VERSION = '1.1.3'.freeze
```

File: app/models/devise_token_auth/concerns/user.rb (hunk 2 above)
Comment (MaicolBen) [reply app/models/devise_token_auth/concerns/user.rb:150]: I think this doesn't make sense anymore as we have this in master https://github.com/lynndylanhurley/devise_token_auth/commit/4a103eddfa503354356704dcd2163e3cfc1a1f16
